MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+/**
+ Emitted when something changed in the `NavigationSettings`.
+ */
+public let NavigationSettingsDidChange = Notification.Name("MBNavigationSettingsDidChange")
+
+/**
+ `NavigationSettings` provides a wrapper for UserDefaults.
+ 
+ Properties are prefixed and before they are stored in UserDefaults.standard.
+ */
+@objc(MBNavigationSettings)
+public class NavigationSettings: NSObject {
+    
+    /**
+     The volume that the voice controller will use.
+     
+     This volume is relative to the system’s volume where 1.0 is same volume as the system.
+    */
+    @objc public dynamic var voiceVolume    : Float     = 1.0
+    
+    /**
+     Indicates whether the voice controller should be muted or not.
+     */
+    @objc public dynamic var voiceMuted     : Bool      = false
+    
+    
+    public static let shared = NavigationSettings()
+    
+    lazy var properties: [Mirror.Child] = {
```

Comment:
I'm usually not a big fan of using `Mirror` in production code, but I see why you used it here. Since this is such an unusual pattern, can you add some doc here explaining what `properties` does and the mechanism of action at play in this class?